docs/developers-docs/development-environment/frontend_environment.md

```diff
@@ -0,0 +1,160 @@
+{: .no_toc }
+
+1. TOC
+{:toc}
+
+## Fork and clone the UI source code
+
+First fork the [ontoportal_web_ui](https://github.com/ontoportal/ontoportal_web_ui) repository. More details about how to can be found in the [Github official doc](https://docs.github.com/en/get-started/quickstart/fork-a-repo?platform=linux)
+
+Then clone your forked repository locally
+```bash
+  git clone https://github.com/your_organization/ontportal_web_ui.git
+```
+
+{: .important }
+The `master` branch represent the [Bioportal version](https://bioportal.bioontology.org/), if you want to have the [Agroportal version](https://agroportal.lirmm.fr/).
+You have to merge the `ontoportal-lirmm` branch to your `master`, by doing so `git merge ontoportal-lirmm master`
+
+## Setting up the development config files
+
+### Environment and database variables
+
+{: .highlight }
+A sample file of `.env` can be found
+here:[https://github.com/ontoportal-lirmm/bioportal_web_ui/blob/master/.env.sample](https://github.com/ontoportal-lirmm/bioportal_web_ui/blob/master/.env.sample)
+
+In command line do
+
+```bash
+cp .env.sample .env
+
+cp config/database.yml.sample config/database.yml
+```
+
+Then edit the file `config/database.yml` with your `user` and `password` of your database. If using the docker
+infrastructure use the following `config/database.yml`
+
+```yml
+    development:
+      adapter: mysql2
+      encoding: utf8
+      database: bioportal_ui
+      username: root
+      password: bp_user
+      host: db
+```
+
+### Update the development.rb file
+
+{: .highlight }
+A sample file of `config/environments/development.rb` can be found
+here:[https://github.com/ontoportal/ontoportal_web_ui/blob/master/config/environments/development.rb](https://github.com/ontoportal/ontoportal_web_ui/blob/master/config/environments/development.rb)
+
+Update like below the line in `config/environments/development.rb` file
+
+```ruby
+ # memcache setup
+config.cache_store = ActiveSupport::Cache::MemCacheStore.new('cache', namespace: 'BioPortal')
+
+```
+
+### Create bioportal_config_development.rb file
+
+{: .highlight }
+A sample file of `config/bioportal_config_development.rb` can be found
+here:[https://github.com/ontoportal/ontoportal_web_ui/blob/master/config/bioportal_config_env.rb.sample](https://github.com/ontoportal/ontoportal_web_ui/blob/master/config/bioportal_config_env.rb.sample)
+
+```bash
+cp .env.sample .env
+
+cp bioportal_config_env.rb.sample bioportal_config_development.rb
+```
+
+Update/Add the following lines to `config/bioportal_config_development.rb`  
+
+```ruby
+
+$HOSTNAME = 'localhost' # PUT YOUR SERVER HOSTNAME (IP) HERE
+
+# The URL for the BioPortal Rails UI (this application)
+$UI_URL = "http://#{$HOSTNAME}:3000"
+
+# REST core service address
+$REST_URL = "http://#{$HOSTNAME}:8080"
+$SPARQL_URL = "http://#{$HOSTNAME}:8081/test/"
+
+$BIOMIXER_URL = "http://#{$HOSTNAME}:8081/BioMixer"
+
+$ANNOTATOR_URL = $PROXY_URL = "http://#{$HOSTNAME}:8081/annotator"
```

Comment:
To get this work with default settings I used #{REST_URL} instead of http://#{$HOSTNAME}:8081

Reply:
This applies also to the other lines involving http://#{$HOSTNAME}:8081

Reply:
Did you put your appliance IP in $HOSTAME instead of localhost